tensorflow/core/kernels/cudnn_rnn_ops.cc

```diff
@@ -1145,6 +1360,45 @@ class CudnnRNNKernelCommon : public OpKernel {
   CudnnModelTypes model_types_;
 };
 
+template <>
+Status CudnnRNNKernelCommon::CreateRnnDescriptor<bfloat16>(
+    OpKernelContext* context, const CudnnRnnModelShapes& model_shapes,
+    const RnnInputMode& input_mode, const AlgorithmConfig& algo_config,
+    ScratchAllocator* dropout_state_allocator,
+    std::unique_ptr<RnnDescriptor>* rnn_desc, bool use_padded_io) {
+  StreamExecutor* executor = context->op_device_context()->stream()->parent();
+  se::dnn::DataType data_type = ToDataType<float>::value;
```

Comment:
Can we use the `std::conditional_t` in the original template instead of the template specialization to better express the intention (and minimize the code duplicate)? like:

```c++
// Some comments
constexpr bool is_bf16 = std::is_same_v<T, bfloat16>;
using Coefficient = std::conditional_t<is_bf16, float, T>;
se::dnn::DataType data_type = ToDataType<Coefficient>::value;
```

Same for the `GetCachedRnnDescriptor`.

Reply:
Thanks, done.